src/main/java/jcifs/http/NtlmHttpURLConnection.java

```diff
@@ -0,0 +1,827 @@
+                values.add(value);
+                break;
+            }
+        }
+        if ( values == null ) {
+            values = new ArrayList<>();
+            values.add(value);
+            this.requestProperties.put(key, values);
+        }
+        // 1.3-compatible.
+        StringBuffer buffer = new StringBuffer();
+        Iterator<String> propertyValues = values.iterator();
+        while ( propertyValues.hasNext() ) {
+            buffer.append(propertyValues.next());
+            if ( propertyValues.hasNext() ) {
+                buffer.append(", ");
+            }
+        }
+        this.connection.setRequestProperty(key, buffer.toString());
+    }
+
+
+    @Override
+    public String getRequestProperty ( String key ) {
+        return this.connection.getRequestProperty(key);
+    }
+
+
+    @Override
+    public Map<String, List<String>> getRequestProperties () {
+        Map<String, List<String>> map = new HashMap<>();
+        for ( Entry<String, List<String>> entry : this.requestProperties.entrySet() ) {
+            map.put(entry.getKey(), Collections.unmodifiableList(entry.getValue()));
+        }
+        return Collections.unmodifiableMap(map);
+    }
+
+
+    @Override
+    public void setInstanceFollowRedirects ( boolean instanceFollowRedirects ) {
+        this.connection.setInstanceFollowRedirects(instanceFollowRedirects);
+    }
+
+
+    @Override
+    public boolean getInstanceFollowRedirects () {
+        return this.connection.getInstanceFollowRedirects();
+    }
+
+
+    @Override
+    public void setRequestMethod ( String requestMethod ) throws ProtocolException {
+        this.connection.setRequestMethod(requestMethod);
+        this.method = requestMethod;
+    }
+
+
+    @Override
+    public String getRequestMethod () {
+        return this.connection.getRequestMethod();
+    }
+
+
+    @Override
+    public int getResponseCode () throws IOException {
+        handshake();
+        return this.connection.getResponseCode();
+    }
+
+
+    @Override
+    public String getResponseMessage () throws IOException {
+        handshake();
+        return this.connection.getResponseMessage();
+    }
+
+
+    @Override
+    public void disconnect () {
+        this.connection.disconnect();
+        this.handshakeComplete = false;
+        this.connected = false;
+    }
+
+
+    @Override
+    public boolean usingProxy () {
+        return this.connection.usingProxy();
+    }
+
+
+    @Override
+    public InputStream getErrorStream () {
+        handshake();
+        return this.connection.getErrorStream();
+    }
+
+
+    private int parseResponseCode () throws IOException {
+        try {
+            String response = this.connection.getHeaderField(0);
+            int index = response.indexOf(' ');
+            while ( response.charAt(index) == ' ' )
+                index++;
+            return Integer.parseInt(response.substring(index, index + 3));
+        }
+        catch ( Exception ex ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)